crates/templates/src/context.rs

```diff
@@ -105,21 +106,72 @@ pub trait TemplateContext: Serialize {
     ///
     /// This is then used to check for template validity in unit tests and in
     /// the CLI (`cargo run -- templates check`)
-    fn sample(now: chrono::DateTime<Utc>, rng: &mut impl Rng, locales: &[DataLocale]) -> Vec<Self>
+    fn sample(
+        now: chrono::DateTime<Utc>,
+        rng: &mut impl Rng,
+        locales: &[DataLocale],
+    ) -> BTreeMap<SampleIdentifier, Self>
     where
         Self: Sized;
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct SampleIdentifier {
```

Comment:
Hmm, this feels like it could be a `Vec<(Key, Value)>` instead? (with `Key` probably being `&'static str` and `Value` being a `String`?)

and then the various stages push in that stack, e.g. at the end you get `[("lang", "en"), ("user", "alice"), ("index", "1")]`, and when you reconstruct the file name you get `lang:en-user:alice-index:1`?

On why a Vec and not a btreemap: it allows duplicate key, and probably that the various levels of encapsulation will have some semantic meaning

Reply:
Have changed to this approach. Example filename is `pages_upstream_oauth2_suggest_link-index:0-browser-session:0-locale:fr.html`. Are we happy with that?

Though I am not sure `:` is allowed on Windows, maybe it'd be better to find something else.